indigo_api/templatetags/indigo.py

```diff
@@ -19,3 +19,49 @@ def work_resolver_url(context, work):
         frbr_uri = '/akn' + frbr_uri
 
     return context.get('resolver_url', settings.RESOLVER_URL) + frbr_uri
+
+
+@register.simple_tag
+def commenced_provisions_description(document, commencement, uncommenced=False):
+    work = document.work
+    if uncommenced:
+        uncommenced_provisions = document.work.uncommenced_provisions()
+        # TODO: same as below for uncommenced provisions
+        return ', '.join([p.id for p in uncommenced_provisions])
+
+    commenceable_provisions = work.commenceable_provisions()
+    commenced_provisions = commencement.provisions
+    runs = []
+    current_run = []
+
+    def end_current_run(run):
+        if len(run) < 4:
+            run = [f'{item["type"]} {item["num"]}' for item in run]
+        else:
+            run = [f'{run[0]["type"]}s {run[0]["num"]} to {run[-1]["num"]}']
+        runs.append(run)
+
+    for i, p in enumerate(commenceable_provisions):
+        if i == 0:
+            if p.id in commenced_provisions:
+                current_run.append({'type': p.type, 'num': p.num.split('.')[0] if p.num else ''})
```

Comment:
The `split` will bork if the number is `1.2.3` - instead, how about `p.num.strip('.')`

Reply:
I *think* there's a slightly simpler way of managing the different cases that reduces the amount of code. But it's academic.